src/book/02-system/02-subsystems/04-motion.mdx

```diff
@@ -5,3 +5,85 @@ title: Motion
 description: Motion and how it works in the NUbots codebase.
 slug: /system/subsystems/motion
 ---
+
+The motion subsystem includes modules that send servo targets to the motors. These include walking, kicking, moving the head, and running scripts.
+
+The Script Engine is detailed on this page, while Script Runner and Script Tuner can be found on the [Behaviour](/system/subsystems/behaviour#script-runner) page.
+
+# Script Engine
+
+Scripts are static motions for the robot. They specify what joint angles to move to and how long the robot should take to get to those joint positions. For example, standing up is a basic script telling the robot to move its joints to the stand position over one second. There can be many of these position specifications in a row to make the robot do more complex movements like getting up or kicking.
+
+To learn how to tune scripts, see the [ScriptTuner guide](/guides/main/script-tuner).
+
+## Messages
+
+To execute a script, we emit an "execute script" message describing the script to execute. This message is received in the ScriptEngine, which executes the appropriate script. These are specified in [Script.h](https://github.com/NUbots/NUbots/blob/master/shared/extension/Script.h) as [ExecuteScriptByName](https://github.com/NUbots/NUbots/blob/96357855b059cd499021552b3e25a4b158828dc5/shared/extension/Script.h#L224) and [ExecuteScript](https://github.com/NUbots/NUbots/blob/96357855b059cd499021552b3e25a4b158828dc5/shared/extension/Script.h#L261).
```

Comment:
There is a stack of direct links to lines in these files, and I feel like this'll become quite troublesome if there's ever changes any of these files. It seems like a bad idea to me

Reply:
The alternative is linking to `master`, but the files may be changed or get deleted. This is a permalink that will always point to the same file in that specific commit.

Reply:
Ohhh, I didn't realise that these were pointing to a specific commit! That sorts that issue.

Reply:
That said, the one pointing to `Script.h` is not a permalink 😅